src/main/java/gyro/google/compute/InstanceGroupResource.java

```diff
@@ -0,0 +1,300 @@
+/*
+ * Copyright 2019, Perfect Sense, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gyro.google.compute;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import com.google.api.client.googleapis.json.GoogleJsonResponseException;
+import com.google.api.services.compute.Compute;
+import com.google.api.services.compute.model.InstanceGroup;
+import com.google.api.services.compute.model.InstanceGroupsSetNamedPortsRequest;
+import com.google.api.services.compute.model.Operation;
+import com.google.cloud.compute.v1.ProjectGlobalNetworkName;
+import gyro.core.GyroException;
+import gyro.core.GyroUI;
+import gyro.core.Type;
+import gyro.core.resource.Id;
+import gyro.core.resource.Output;
+import gyro.core.resource.Resource;
+import gyro.core.resource.Updatable;
+import gyro.core.scope.State;
+import gyro.core.validation.Regex;
+import gyro.core.validation.Required;
+import gyro.google.Copyable;
+
+/**
+ * Creates an instance group.
+ *
+ * Examples
+ * --------
+ *
+ * .. code-block:: gyro
+ *
+ *     google::compute-instance-group instance-group-example
+ *          name: "instance-group-example"
+ *          description: "instance-group-example-description"
+ *          zone: "us-central1-a"
+ *     end
+ *
+ * .. code-block:: gyro
+ *
+ *     google::compute-instance-group instance-group-named-ports-example
+ *          name: "instance-group-named-ports-example"
+ *          description: "instance-group-named-ports-example-description"
+ *          zone: "us-central1-a"
+ *
+ *          named-port
+ *              name: "port-a"
+ *              port: 123
+ *          end
+ *
+ *          named-port
+ *              name: "port-a"
+ *              port: 123
+ *          end
```

Comment:
exact same named port, does this actually makes two named port on the instance group ?